oauth2client/client.py

```diff
@@ -1617,6 +1617,23 @@ def _revoke(self, http_request):
         """
         self._do_revoke(http_request, self.access_token)
 
+    def sign_blob(self, blob):
+        """Cryptographically sign a blob (of bytes).
+
+        Args:
+            blob: bytes, Message to be signed.
+
+        Returns:
+            tuple, A pair of the private key ID used to sign the blob and
+            the signed contents.
+        """
+        _abstract()
+
+    @property
+    def service_account_email(self):
```

Comment:
I was on the fence on this one, but this abstract class has three subclasses and they all have a `service_account_email`, so I went with putting it here as a signal that `isinstance(foo, AppAssertionCredentials)` would mean `foo.service_account_email` is there.